display/ssd1306.h

```diff
@@ -225,15 +229,39 @@ class SSD1306 : public I2CDevice, Looper, StateMachine, SaberBase, private Audio
 	}
 	if (eof_) {
 	  // STDOUT << "EOF " << frame_count_ << "\n";
-	  screen_ = SCREEN_PLI;
-	  if (frame_count_ == 1) return 5000;
-	  return FillFrameBuffer();
+    if (!SaberBase::IsOn()) {
+	   screen_ = SCREEN_PLI;
+	    if (frame_count_ == 1) return font_config.ProffieOSImageDuration;
+	    return FillFrameBuffer();
+    }
 	}
 	frame_count_++;
-	if (looped_frames_ > 1 && millis() - loop_start_ > 5000) {
-	  STDOUT << "END OF LOOP\n";
-	  screen_ = SCREEN_PLI;
-	}
+  if (SaberBase::IsOn()) {
+    // Single frame image
+    if (looped_frames_ == 1) {
+      if (frame_count_ == 1) return font_config.ProffieOSImageDuration;
+      if (frame_count_ > 1) {
+        if (SaberBase::Lockup()) {
+          ShowFile(&IMG_lock);
+        } else {
+          screen_ = SCREEN_PLI;
+          if (!imgon_) {
```

Comment:
This, and all of the logic in here needs comments to make it obvious what is going on.

Reply:
I'm actually a little confused, because I don't know how we get here if imgon_ is false.

Reply:
i changed the logic for detecting static on.bmp, it's much cleaner now.